exporter/elasticsearchexporter/data-stream-router.go

```diff
@@ -0,0 +1,67 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package elasticsearchexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// routeLogRecord returns the name of the index to send the log record to according to data stream routing attributes.
+// It searches for the routing attributes on the log record, scope, and resource.
+// It creates missing routing attributes on the log record if they are not found.
+func routeLogRecord(
+	record *plog.LogRecord,
+	scope pcommon.InstrumentationScope,
+	resource pcommon.Resource,
+) string {
+	dataSet := ensureAttribute(dataStreamDataset, defaultDataStreamDataset, record.Attributes(), scope.Attributes(), resource.Attributes())
+	namespace := ensureAttribute(dataStreamNamespace, defaultDataStreamNamespace, record.Attributes(), scope.Attributes(), resource.Attributes())
+	dataType := ensureAttribute(dataStreamType, defaultDataStreamTypeLogs, record.Attributes(), scope.Attributes(), resource.Attributes())
+	return fmt.Sprintf("%s-%s-%s", dataType, dataSet, namespace)
```

Comment:
Not blocking this PR but the type should be fixed based on the record type. Similar story for metrics and spans.

```suggestion
	recordAttributes.PutStr(dataStreamType, "logs")
	return fmt.Sprintf("logs-%s-%s", dataSet, namespace)
```